### DISCUSSION TITLE
Unable to publish after update from 3.8.5 to 5.7.1

### DISCUSSION THREAD

**Discussion by brianpilati** [verdaccio/verdaccio#3071] [Q&A]
Running version 3.8.5 I was able to run a command in a node.js script and publish an npm to a private verdaccio server. After updating to v5.7.1, I can no longer publish to my local repo. However, I never updated the remote private verdaccio server and I can still publish to that one.

What can I do to continue to use my script on my local private verdaccio server and eventually on the remote server?

### Node Script
const childProcess = exec(`npm publish --registry http://localhost:4873 ./`);

### config.yaml
auth:
  htpasswd:
    file: ./htpasswd
    # Maximum amount of users allowed to register, defaults to "+inf".
    # You can set this to -1 to disable registration.
    # max_users: 1

packages:
  '@*/*':
    # scoped packages
    access: $all
    publish: $authenticated
    proxy: npmjs

  '**':
    access: $all

    publish: $authenticated

    proxy: npmjs

**Comment by juanpicado** [comment]
Ok wow big change, I would create a way I can reproduce because might be 100 things.

**Reply by brianpilati** [reply]
It's easy to reproduce:

1- Install verdaccio@5.7.1
2- write a nodejs script with exec(npm publish --registry http://localhost:4873 ./);
3- start verdaccio
4- $ npm publish --registry http://localhost:4873 ./
5- Look in logs for "undefined is forbidden"

 warn --- http address - http://localhost:4873/ - verdaccio/5.7.1
 http --- 127.0.0.1 requested 'PUT /@npm%2fname'
 error--- undefined is forbidden publish for @npm/name
 http --- 200, user: null(127.0.0.1), req: 'PUT /@npm%2fname', bytes: 4179/0

The same script always publishes to an remote private server running verdaccio@3.8.5 with this command
$ npm publish --registry http://x.x.x.x:4873 ./

**Reply by juanpicado** [reply]
Have you read this https://verdaccio.org/blog/2019/02/24/migrating-verdaccio?

**Reply by brianpilati** [reply]
I reviewed the information. I did update my script to include the following: 

const childProcess = exec(`export VERDACCIO_USER_NAME=${verdaccioUsername};  npm publish --registry http://localhost:4873/ ./);

No change.

Do you have anything more specific from that document?

**Reply by brianpilati** [reply]
@juanpicado Any ideas? I am still stuck.

This is not in the logs but on the output on the console:

npm ERR! Unable to authenticate, your authentication token seems to be invalid.

I have looked in the .npmrc file and I have an _authToken for //localhost:4873

and I have an entry in the .htpasswd file for my user

Thoughts?

**Reply by juanpicado** [reply]
You would need give me some time 🙏🙏 but I will do my best to give some try this weekend, I haven't forgotten you. We will solve it.

**Reply by brianpilati** [reply]
@juanpicado Thanks. Good luck!

**Reply by juanpicado** [reply]
Looking into detail your issue, I see

```
warn --- http address - http://localhost:4873/ - verdaccio/5.7.1
http --- 127.0.0.1 requested 'PUT /https://github.com/npm%2fname'
error--- undefined is forbidden publish for @npm/name
```
When you are running ´exec(npm publish --registry http://localhost:4873/ ./);´ I winder which npm version is, because seems is not sending the token.

The token is need it to autenticate the request, that is the reason why it fails I am not sure why works on v3 ( I not longer support that old version) but the error is quite clear what is missing.

**Reply by brianpilati** [reply]
@juanpicado I understand that is what is happening. Now how do I fix it? How do I add my token? What has changed between v3 and v5 regarding the token?

I am running npm 8.5.5
